File: P1S3_JAVA/src/GestorFiltros.java
public class GestorFiltros {
    Target ministerioSanidad;
    CadenaFiltros cadenaFiltros;
    Formulario formulario;

    public GestorFiltros(){
        ministerioSanidad = new Target();
        cadenaFiltros = new CadenaFiltros();
        formulario = new Formulario();
    }

    public GestorFiltros(Target t){
        ministerioSanidad = t;
        cadenaFiltros = new CadenaFiltros();
        formulario = new Formulario();
    }

    public void setFormulario(Formulario formulario) {
        this.formulario = formulario;
    }

    void addFiltros(Filtro f){
        cadenaFiltros.addFiltros(f);
    }

    void deleteFiltros(Filtro f){
        cadenaFiltros.deleteFiltros(f);
    }

    void procesarFormulario(){
        cadenaFiltros.procesarFormulario(formulario);
        enviarFormulario();
    }

    void enviarFormulario(){
        ministerioSanidad.recibirFormulario(formulario);
    }
}

File: P1/P1S3/GestorFiltros.cpp
//
// Created by ines on 8/4/21.
//

#include "GestorFiltros.h"

GestorFiltros::GestorFiltros() {}

GestorFiltros::GestorFiltros(Target t) {
    ministerioSanidad = t;
}

void GestorFiltros::setFormulario(Formulario f) {
    formulario = f;
}

void GestorFiltros::addFiltros(Filtro *f) {
    cadenaFiltros.addFiltros(f);
}

void GestorFiltros::deleteFiltros(Filtro *f) {
    cadenaFiltros.deleteFiltros(f);
}

//Método para ejecutar el FilterManager
void GestorFiltros::procesarFormulario() {
    cadenaFiltros.procesarFormulario(formulario);
    enviarFormulario();
}

void GestorFiltros::enviarFormulario() {
    ministerioSanidad.recibirFormulario(formulario);
}